packages/php-wasm/compile/php-wasm-memory-storage/wasm_memory_storage.c

```diff
@@ -0,0 +1,87 @@
+/* wasm_memory_storage extension for PHP */
+
+#ifdef HAVE_CONFIG_H
+# include "config.h"
+#endif
+
+#include <stdio.h>
+#include "php.h"
+#include "ext/standard/info.h"
+#include "Zend/zend_alloc.h"
+#include "php_wasm_memory_storage.h"
+
+void* wasm_memory_storage_chunk_alloc(zend_mm_storage* storage, size_t size, size_t alignment)
+{
+	void* ptr = NULL;
+	if (posix_memalign(&ptr, alignment, size) == 0)
```

Comment:
@dmsnell which version of the man page?

In the man pages I looked at on the web, I'm not seeing such a word of caution. I hope that is a hint that it is probably OK.

It would be interesting to read how Emscripten allocators achieve the alignment. I can skim that later today for any hint of additional storage requirements. But I don't see why there would need to be. It is just allocated memory that can be tracked the same way as everything else, I would think. ...Unless there is memory used to make future aligned allocations faster or something.

Reply:
@brandonpayton both the `man 3 posix_memalign` page on my macOS install plus the one downloaded into my docs browser share that message in the **Notes** section.

kind of makes me wonder if the posix version isn't doing something similar to what PHP does. maybe it's not faster? maybe it's just automatic.

Reply:
👍   I was mobile before and was just searching. Hadn't checked the macOS one yet :)  Thank you for bringing this up.

Regarding speed, I did some semi-naive perf testing, and with 1000 sequential requests, it took 8 seconds with this change and 9 seconds without it. I can share details later in a separate comment.

Reply:
sounds close enough that it doesn't matter